relay/metadata_provider.go

```diff
@@ -119,8 +127,7 @@ func (m *metadataProvider) GetMetadataForBlobs(keys []v2.BlobKey) (metadataMap,
 		}()
 	}
 
-	mMap := make(metadataMap)
-	for len(mMap) < len(keys) {
```

Comment:
This hangs if there are duplicates in `keys`

Reply:
Add a unit test for this

Reply:
Done